common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -396,7 +389,18 @@ public boolean startsWith(final UTF8String prefix, int collationId) {
     if (collationId == CollationFactory.LOWERCASE_COLLATION_ID) {
       return this.toLowerCase().startsWith(prefix.toLowerCase());
     }
-    return matchAt(prefix, 0, collationId);
+    return collatedStartsWith(prefix, collationId);
+  }
+
+  private boolean collatedStartsWith(final UTF8String prefix, int collationId) {
```

Comment:
Could you point out which unit tests in `UTF8StringSuite` check the functions.

Reply:
I think we test these functions only in CollationSuite now, as they are only used in context of collations anyways. To remove clutter, we opted to remove unit tests from `UTF8StringSuite` and `StringExpressionSuite` because those in `CollationSuite` seem to do just fine

Reply:
I think that @MaxGekk has a point.
`CollationSuite` becomes a bit over cluttered and it should be used for for E2E testing. Can we either add unit test to `UTF8StringSuite` or create `UTF8StringSuiteWithCollation` suite?

Option is also to use `CollationExpressionSuite`.

I would propose following:
1) E2E collation tests go to `CollationSuite`.
2) String level tests go to either `UTF8StringSuite` or `UTF8StringWithCollationSuite`.
3) Expression level tests go to `CollationExpressionSuite`.
4) Collation management tests to to `CollationFactorySuite`.

@MaxGekk and @cloud-fan - what are your thought on this?

Reply:
SGTM, and +1 to `UTF8StringWithCollationSuite`